ChangeLog.d/selftest-gcc12.txt

```diff
@@ -0,0 +1,2 @@
+Bugfix
+   * Silence a warning from GCC 12 in the selftest program. Fixes #5974.
```

Comment:
Hmm, when I read "silence a warning" I expect the functionality to remain the same but expressed in a different way

What do you think about

```suggestion
   * Remove code in the selftest program that caused a bogus use-after-free
     warning in GCC 12. Fixes #5974.
```

Reply:
(I'd understand if there isn't time to change this now)